test/cmocka/src/math/matrix/ref_matrix.m

```diff
@@ -0,0 +1,186 @@
+% ref_matrix - Generate C header files for matrix library unit tests
+
+% SPDX-License-Identifier: BSD-3-Clause
+%
+% Copyright(c) 2022 Intel Corporation. All rights reserved.
+
+function ref_matrix()
+	opt.test_n = 1;
+	opt.input = 'random';
+	opt.a_rows = 3;
+	opt.a_columns = 4;
+	opt.b_rows = 4;
+	opt.b_columns = 5;
+	opt.a_qxy_x = 1;
+	opt.a_qxy_y = 15;
+	opt.b_qxy_x = 1;
+	opt.b_qxy_y = 15;
+	opt.c_qxy_x = 2;
+	opt.c_qxy_y = 14;
+	opt.elementwise = 0;
+	get_ref_mult(opt);
+
+	opt.test_n = 2;
+	opt.input = 'random';
+	opt.a_rows = 6;
+	opt.a_columns = 7;
+	opt.b_rows = 7;
+	opt.b_columns = 8;
+	opt.a_qxy_x = 7;
+	opt.a_qxy_y = 0;
+	opt.b_qxy_x = 7;
+	opt.b_qxy_y = 0;
+	opt.c_qxy_x = 16;
+	opt.c_qxy_y = 0;
+	opt.elementwise = 0;
+	get_ref_mult(opt);
+
+	opt.test_n = 3;
+	opt.input = 'random';
+	opt.a_rows = 5;
+	opt.a_columns = 6;
+	opt.b_rows = 5;
+	opt.b_columns = 6;
+	opt.a_qxy_x = 1;
+	opt.a_qxy_y = 15;
+	opt.b_qxy_x = 1;
+	opt.b_qxy_y = 15;
+	opt.c_qxy_x = 1;
+	opt.c_qxy_y = 15;
+	opt.elementwise = 1;
+	get_ref_mult(opt);
+
+	opt.test_n = 4;
+	opt.input = 'random';
+	opt.a_rows = 5;
+	opt.a_columns = 1;
+	opt.b_rows = 1;
+	opt.b_columns = 4;
+	opt.a_qxy_x = 1;
+	opt.a_qxy_y = 15;
+	opt.b_qxy_x = 1;
+	opt.b_qxy_y = 15;
+	opt.c_qxy_x = 1;
+	opt.c_qxy_y = 15;
+	opt.elementwise = 0;
+	get_ref_mult(opt);
+end
+
+function get_ref_mult(opt)
+	[a_bits, a_scale] = get_qxy_scale(opt.a_qxy_x, opt.a_qxy_y);
+	[b_bits, b_scale] = get_qxy_scale(opt.b_qxy_x, opt.b_qxy_y);
+	[c_bits, c_scale] = get_qxy_scale(opt.c_qxy_x, opt.c_qxy_y);
+	bits = max([a_bits b_bits c_bits]);
+	if (a_bits ~= b_bits)
+		error('Matrix word lengths need to be the same (for now)');
+	end
+
+	% Data to process
+	switch lower(opt.input)
+		case 'ones'
+			a = ones(opt.a_rows, opt.a_columns);
+			b = ones(opt.b_rows, opt.b_columns);
+		case 'random'
+			a = (rand(opt.a_rows, opt.a_columns) * 2 - 1) * a_scale;
+			b = (rand(opt.b_rows, opt.b_columns) * 2 - 1) * b_scale;
+		otherwise
+			error('Illegal input');
+	end
+
+	[iqa, fqa] = quant_qxy(a, opt.a_qxy_x, opt.a_qxy_y);
```

Comment:
Can we use data types like int8, int16 etc instead of double?

Reply:
Yep, it could be done if it works OK in Octave. I'll test.

Reply:
It worked, the quantize returns now fixed type.